tests/scripts/restore-mon-quorum-using-osds.sh

```diff
@@ -0,0 +1,319 @@
+
+        if [ ${desired_count} = ${actual_count} ]; then
+           break
+        fi
+
+        if [ "${retry}" = "${RETRY_COUNT}" ] ; then
+          echo 'timed out'
+            exit 1
+        fi
+        sleep "${RETRY_PERIOD}"
+        retry=$((retry+1))
+    done
+}
+
+function restore_manifests() {
+    BACKUP_MON_DEPLOYMENTS=$(ls "${BACKUP_DIR}"/rook-ceph-mon-*.yaml 2>/dev/null)
+    DESIRED_MON_COUNT=$(echo "${BACKUP_MON_DEPLOYMENTS}" | wc -w)
+
+    for mon in ${BACKUP_MON_DEPLOYMENTS}; do
+        kubectl replace --force -f "${mon}"
+    done
+    for i in ${MON_IDS}; do
+        kubectl delete -n "${NAMESPACE}" pod -l app=rook-ceph-mon -l mon="${i}"
+    done
+    wait_pods rook-ceph-mon "${DESIRED_MON_COUNT}" Running
+
+    BACKUP_OSD_DEPLOYMENTS=$(ls "${BACKUP_DIR}"/rook-ceph-osd-*.yaml 2>/dev/null)
+    DESIRED_OSD_COUNT=$(echo "${BACKUP_OSD_DEPLOYMENTS}" | wc -w)
+
+    for osd in ${BACKUP_OSD_DEPLOYMENTS}; do
+        kubectl replace --force -f "${osd}"
+    done
+    for i in ${OSD_IDS}; do
+        kubectl delete -n "${NAMESPACE}" pod -l app=rook-ceph-osd -l osd="${i}"
+    done
+
+    BACKUP_TOOL_DEPLOYMENT=$(ls "${BACKUP_DIR}"/rook-ceph-tools.yaml 2>/dev/null)
+    kubectl replace --force -f "${BACKUP_TOOL_DEPLOYMENT}"
+    kubectl -n "${NAMESPACE}" delete pod -l app=rook-ceph-tools
+    wait_pods rook-ceph-tools 1 Running
+    TOOL=$(kubectl get -n "${NAMESPACE}" pod -l app=rook-ceph-tools -o json | jq -r '.items[0].metadata.name')
+}
+
+function restore() {
+    for i in ${MON_IDS}; do
+        echo "Restoring monitor store of mon ${i}"
+        name=$(kubectl get -n "${NAMESPACE}" pod -l app=rook-ceph-mon -l mon="${i}" -o json | jq -r '.items[].metadata.name')
+        STORE_DB=/var/lib/ceph/mon/ceph-"${i}"/store.db
+        kubectl exec -i -n "${NAMESPACE}" "${name}" -- bash -c "test -d ${STORE_DB}.corrupted && rm -rf ${STORE_DB} && mv ${STORE_DB}.corrupted ${STORE_DB} || true"
+    done
+    restore_manifests
+
+    kubectl -n "${NAMESPACE}" scale deployment rook-ceph-operator --replicas=1
+    kubectl -n "${NAMESPACE}" delete pod -l app=rook-ceph-operator
+    wait_pods rook-ceph-operator 1 Running
+
+    DESIRED_MGR_COUNT=$(kubectl get deployment -n "${NAMESPACE}" -l app=rook-ceph-mgr --no-headers 2>/dev/null | wc -l)
+    kubectl -n "${NAMESPACE}" delete pod -l app=rook-ceph-mgr
+    wait_pods rook-ceph-mgr "$DESIRED_MGR_COUNT" Running
+}
+
+function init() {
+    if ! command -v kubectl; then
+        echo "'kubectl' must exist" >&2
+        exit 1
+    fi
+    if ! command -v jq;  then
+        echo "'jq' must exist" >&2
+        exit 1
+    fi
+    if ! kubectl get pod >/dev/null 2>&1; then
+        echo "couldn't connect to your kubernetes cluster" >&2
+        exit 1
+    fi
+
+    TOOL_COUNT=$(kubectl get pod -n "${NAMESPACE}" -l app=rook-ceph-tools --no-headers 2>/dev/null | wc -l)
+    if [ "${TOOL_COUNT}" != 1 ]; then
```

Comment:
Is this working for you? I get this comparison error for the toolbox existing, even though it exists:
```
++ kubectl get pod -n rook-ceph -l app=rook-ceph-tools --no-headers
++ wc -l
+ TOOL_COUNT='       1'
+ '[' '       1' '!=' 1 ']'
+ echo 'toolbox pod must exist'
toolbox pod must exist
```

Reply:
@travisn 

> Is this working for you? I get this comparison error for the toolbox existing, even though it exists:

Yes. In my case, TOOL_COUNT is not '       1' but '1'. I'm not sure why it makes such a difference, but I changed the `if`statement as follows and it should fix this problem.

```
- if [ "${TOOL_COUNT}" != 1 ]; then
+  if [ ${TOOL_COUNT} != 1 ]; then
````

I also removed similar double quotes.

Could you try the new version? Sorry to bother you.